bindings/python/CMakeLists.txt

```diff
@@ -78,7 +78,7 @@ if ( NOT ${VALID_PIP_EXIT_CODE} EQUAL 0 )
   install(
     CODE
     "EXECUTE_PROCESS(
-      COMMAND ${PYTHON_EXECUTABLE} ${SETUP_PY} install \
+    COMMAND /usr/bin/env CC=c++ CFLAGS=\"-std=c++17\" ${PYTHON_EXECUTABLE} ${SETUP_PY} install \
```

Comment:
You definitely do not want to hardcode this anywhere. This should be set through CMake at build configure time with the `CMAKE_C_FLAGS` and `CMAKE_CXX_FLAGS`. c.f. https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_FLAGS.html for more details.

Reply:
Can probably just do something similar to [what ROOT does with `CMAKE_CXX_STANDARD`](https://github.com/root-project/root/blob/a17c8472ee6257a2937db46e465e43cdc6f71638/cmake/modules/CheckCompiler.cmake#L144-L169).

Reply:
You misunderstood my point. The issue is the setup.py which decides to go for the CC / CFLAGS in order to compile a C++11 python module. Notice this affects also some ubuntu versions.

Reply:
I can change it to use CMAKE_CXX_FLAGS, but I think something fishy is going on.

Reply:
@ktf I would suggest that you make a GitHub Issue to discuss this and to clearly describe what the Issue is, what you want to have happen, what in the codebase needs to get changed, and what you plan to implement. 

Also, as @adriansev already pointed out, your PR is targeting the wrong branch. If you refactor this PR to provide a solution that has been agreed upon in the GitHub issue then you should have it target the `master` branch.

Reply:
@matthewfeickert  i did https://github.com/xrootd/xrootd/issues/1660 to discuss there the problem. thanks a lot!